text/0042-extrinsics-state-version.md

```diff
@@ -0,0 +1,104 @@
+| **Description** | Add System Version and remove StateVersion |
+| **Authors**     | Vedhavyas Singareddi                       |
+
+## Summary
+
+At the moment, we have `system_version` field on `RuntimeVersion` that derives which state version is used for the
+Storage.
+We have a use case where we want extrinsics root is derived using `StateVersion::V1`. Without defining a new field
+under `RuntimeVersion`,
+we would like to propose adding `system_version` that can be used to derive both storage and extrinsic state version.
+
+## Motivation
+
+Since the extrinsic state version is always `StateVersion::V0`, deriving extrinsic root requires full extrinsic data.
+For `Subspace` project, we have an enshrined rollups called `Domain` with optimistic verification and Fraud proofs are
+used to detect malicious behavior.
+One of the `Fraud proof` variant is to derive `Domain` block extrinsic root on `Subspace`'s consensus chain.
+Since `StateVersion::V0` requires full extrinsic data, we are forced to pass all the extrinsics through the Fraud proof.
+One of the main challenge here is some extrinsics could be big enough that this variant of Fraud proof may not be
+included in the Consensus block due to Block's weight restriction.
+If the extrinsic root is derived using `StateVersion::V1`, then we do not need to pass the full extrinsic data but
+rather at maximum, 32 byte of extrinsic data.
+
+## Stakeholders
+
+- Technical Fellowship, in its role of maintaining system runtimes.
+
+## Explanation
+
+In order to use project specific StateVersion for extrinsic roots, we proposed
+an [implementation](https://github.com/paritytech/polkadot-sdk/pull/1691) that introduced
+parameter to `frame_system::Config` but that unfortunately did not feel correct.
+So we would like to [propose](https://github.com/paritytech/polkadot-sdk/pull/1968) adding this change to
+the `RuntimeVersion`
+object. The system version, if introduced, will be used to derive both storage and extrinsic state version.
+If system version is `1`, then Storage State version would V1 and Extrinsic State version would use V0.
+If system version is `2`, then both Storage and Extrinsic State version would V1.
+
+If implemented, the new `RuntimeVersion` definition would look something similar to
+
+```rust
+/// Runtime version (Rococo).
+#[sp_version::runtime_version]
+pub const VERSION: RuntimeVersion = RuntimeVersion {
+		spec_name: create_runtime_str!("rococo"),
+		impl_name: create_runtime_str!("parity-rococo-v2.0"),
+		authoring_version: 0,
+		spec_version: 10020,
+		impl_version: 0,
+		apis: RUNTIME_API_VERSIONS,
+		transaction_version: 22,
+		system_version: 1,
+	};
+```
+
+## Drawbacks
+
+Since the `CoreApi` version would need to be bumped, there would require a Runtime upgrade for existing projects
+with `system_version` set to `1` so that it wont brick their chain. New projects can choose between `1` or `2`
+depending on their use-case.
```

Comment:
I don't see why `CoreApi` would need to be bumped. This whole paragraph seems obsolete.

Reply:
Should be corrected now